src/java.base/share/classes/java/util/Timer.java

```diff
@@ -537,6 +565,16 @@ private void mainLoop() {
                             continue;  // No action required, poll queue again
                         }
                         currentTime = System.currentTimeMillis();
+                        /*[IF CRIU_SUPPORT]*/
+                        if (task.criuAdjustRequired) {
+                            long checkpointRestoreTimeDelta = InternalCRIUSupport.getCheckpointRestoreNanoTimeDelta();
+                            if (checkpointRestoreTimeDelta > 0) {
```

Comment:
we should assert if time goes backwards rather than fail sliently

Reply:
Add `throw new InternalError("InternalCRIUSupport.getCheckpointRestoreNanoTimeDelta() returns a negative value " + checkpointRestoreTimeDelta)`